user-interface/src/search-results/SearchResults.test.tsx

```diff
@@ -0,0 +1,237 @@
+  });
+
+  test('should show the no results alert when no results are available', async () => {
+    vi.spyOn(Api2, 'searchCases').mockResolvedValue(buildResponseBodySuccess([]));
+
+    renderWithProps();
+
+    let table = document.querySelector('#search-results > table');
+    expect(table).not.toBeInTheDocument();
+    let noResultsAlert = document.querySelector('#no-results-alert');
+    expect(noResultsAlert).not.toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(document.querySelector('.loading-spinner')).not.toBeInTheDocument();
+      table = document.querySelector('.search-results table');
+      expect(table).not.toBeInTheDocument();
+      noResultsAlert = document.querySelector('#no-results-alert');
+      expect(noResultsAlert).toBeInTheDocument();
+      expect(noResultsAlert).toBeVisible();
+    });
+  });
+
+  test('should show the error alert when an error is encountered', async () => {
+    vi.spyOn(Api2, 'searchCases').mockRejectedValue(
+      buildResponseBodyError(new Error('some error')),
+    );
+
+    renderWithProps();
+
+    let table = document.querySelector('.search-results table');
+    expect(table).not.toBeInTheDocument();
+
+    let searchErrorAlert = document.querySelector('#search-error-alert');
+    expect(searchErrorAlert).not.toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(document.querySelector('.loading-spinner')).not.toBeInTheDocument();
+      table = document.querySelector('#search-results > table');
+      expect(table).not.toBeInTheDocument();
+      searchErrorAlert = document.querySelector('#search-error-alert');
+      expect(searchErrorAlert).toBeInTheDocument();
+      expect(searchErrorAlert).toBeVisible();
+    });
+  });
+
+  test('should render pagination', async () => {
+    renderWithProps();
+
+    await waitFor(() => {
+      expect(document.querySelector('.loading-spinner')).toBeInTheDocument();
+    });
+
+    await waitFor(() => {
+      const pagination = document.querySelector('.usa-pagination');
+      expect(pagination).toBeInTheDocument();
+      expect(pagination).toBeVisible();
+    });
+
+    const nextPage = screen.getByTestId('pagination-button-next-results');
+    fireEvent.click(nextPage);
+
+    await waitFor(() => {
+      expect(document.querySelector('.loading-spinner')).toBeInTheDocument();
+    });
+
+    await waitFor(() => {
+      const pagination = document.querySelector('.usa-pagination');
+      expect(pagination).toBeInTheDocument();
+      expect(pagination).toBeVisible();
+    });
+  });
+
+  test('should not search for an invalid predicate', async () => {
+    renderWithProps({
+      searchPredicate: {
+        limit: 25,
+        offset: 0,
+      },
+    });
+
+    await waitFor(() => {
+      expect(document.querySelector('.loading-spinner')).not.toBeInTheDocument();
+    });
+  });
+});
+
+describe('test sorting functions', () => {
+  test('should sort objects properly by dateFiled', () => {
+    const unsortedDates = [
```

Comment:
move these 2 constants into before each?

Reply:
sure

Reply:
I was actually thinking that if these sort functions may become useful on any other screens, we move them and their tests to their own utility files.

Reply:
Turns out we weren't using them and don't need them.  removed.